lib/constructBodyDigestSignature.test.js

```diff
@@ -0,0 +1,109 @@
+'use strict'
+
+const { test } = require('tap')
+const constructBodyDigestSignature = require('./constructBodyDigestSignature')
+
+test('Constructs a matching Digest header from a valid request Digest header and body', async ({
+  equal
+}) => {
+  const tests = [
+    {
+      description: 'Matching Digest header constructed from an incoming request - using self generated values',
+      input: {
+        headers: {
+          digest: 'sha-256=k6I5cakU5erL8KjSUVTNownDwccvu5kU1Hxg88toFYg=',
+          'content-type': 'application/json'
+        },
+        body: { hello: 'world' }
+      },
+      expected: 'sha-256=k6I5cakU5erL8KjSUVTNownDwccvu5kU1Hxg88toFYg='
+    },
+    {
+      description: 'Matching Digest header constructed from an incoming request - using self generated values',
+      input: {
+        headers: {
+          digest: 'sha-512=+PtokCNHosgo04ww4cNhd4yJxhMjLzWjDAKtKwQZDT4Ef9v/PrS/+BQLX4IX5dZkUMK/tQo7Uyc68RkhNyCZVg==',
+          'content-type': 'application/json'
+        },
+        body: { hello: 'world' }
+      },
+      expected: 'sha-512=+PtokCNHosgo04ww4cNhd4yJxhMjLzWjDAKtKwQZDT4Ef9v/PrS/+BQLX4IX5dZkUMK/tQo7Uyc68RkhNyCZVg=='
+    },
+    {
+      description: 'Matching Digest header constructed from an incoming request - Digest constructed from Jenkins',
+      input: {
+        headers: {
+          digest: 'SHA-512=sLhuceAzAMbd17PdLQUQ7lY50I4PJ1XmZF37Mt3L1GmAKfYYx/GbqevL9A1bT3D+AbQd1D3/EVQtGjGorVcIvA==',
+          'content-type': 'application/json'
+        },
+        body: {
+          project: null,
+          owner: 'telus',
+          branch: null,
+          risk_inputs: [{ input_name: 'embargo', input_args: {} }],
+          webhooks: { chatops: '<test-url>' }
+        }
```

Comment:
We don't want the have this specific example I dont think.

Reply:
Good point. This is the first test that verified the implementation with outside values but, now that I've confirmed the implementation works, the constructed examples (e.g. `{ hello: 'world' }`) should be enough. I've removed this test case.